### PR TITLE
Automatically pass through all on* event attributes

### PR DESCRIPTION
**Pull request by BalusC**
https://github.com/jakartaee/faces/issues/1507

With this PR, the following finally works for any on* event attribute and f:ajax event="..." without need to register event name as a new attribute in component as long as the event name is covered by ClientBehaviorHolder#getEventNames():

```
<h:inputText oninput="...">
    <f:ajax event="input" ... />
</h:inputText>
```

Done so far:
- New `jakarta.faces.component.html.HtmlEvents` helper class for managing all these attributes
- Mojarra impl has been adjusted
- updated `ClientBehaviorHolder#getEventNames()` to catch up [current standard](https://html.spec.whatwg.org/multipage/webappapis.html#event-handlers-on-elements,-document-objects,-and-window-objects) (without the need to add new attribute getter/setter pairs!)
- add context param which makes it possible to add new event names so it doesn't anymore throw exception on them: `jakarta.faces.ADDITIONAL_HTML_EVENT_NAMES`, defined in new `jakarta.faces.component.html.HtmlEvents` class
- set in stone in Faces spec: https://github.com/eclipse-ee4j/mojarra/pull/5325/commits/7236717db399cb08aeb23331c5f16d83911e8913

### CODE REVIEW

**Comment by tandraschko** [comment]
looks good for me

@volosied @melloware can you also give a look plz?

File: impl/src/main/java/jakarta/faces/component/html/HtmlEvents.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+    /**
+     * Events supported by all HTML document elements.
+     */
+    public enum HtmlDocumentElementEvent {
+        abort,
+        auxclick,
+        beforeinput,
+        beforematch,
+        eforetoggle,
```
Comment (melloware) [inline impl/src/main/java/jakarta/faces/component/html/HtmlEvents.java:37]: Spelling mistake should be `beforeToggle` ?

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by melloware** [comment]
I like it.  This is a big change but it will be a good one for 5.0.  I will have to study the MyFaces code on exactly where to put this stuff.

File: impl/src/main/java/jakarta/faces/component/html/HtmlEvents.java (hunk 1 above)
Comment (BalusC) [reply impl/src/main/java/jakarta/faces/component/html/HtmlEvents.java]: Eew, well spotted, fixed now and thanks!

**Comment by volosied** [comment]
Sorry, didn't see the notifications until now. I'll take a look at this tomorrow or Tuesday!

**Comment by BalusC** [comment]
API part of original PR is split into https://github.com/jakartaee/faces/pull/1880

**Review by mnriem — APPROVED** [review]

**Pull request merged**